### PR TITLE
Bug adding another method and test to xseed causes other tests to fail.

### PR DESCRIPTION
**Pull request by dsentinel**
I added a method xseed.Parser._parse_resp and a test ParserTestCase.test_read_resp.
When my test runs to completion in the test suite, these 2 following tests fail
test_string, and 
test_rotation_to_zne

If I run my test after, it works fine:
```

$ python test_parser.py ParserTestCase.test_string ParserTestCase.test_read_resp
..
----------------------------------------------------------------------
Ran 2 tests in 0.070s

OK
```
If my test is run first, it breaks test_string
```
$ python test_parser.py  ParserTestCase.test_read_resp ParserTestCase.test_string
./Users/lloyd/GitHub/obspy_nrl2inv/obspy/obspy/io/xseed/parser.py:1322: UserWarning: There exist unparsed elements!
  warnings.warn("There exist unparsed elements!")
F
======================================================================
FAIL: test_string (__main__.ParserTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_parser.py", line 102, in test_string
    ("BW.MANZ..EHZ | 200.00 Hz | Streckeisen STS-2/N seismometer | "
AssertionError: Lists differ: [u'Networks:', u'BW (BayernNet... != [u'Networks:', u'BW (BayernNet...

First differing element 5:
u'BW.MANZ..EHZ | 200.00 Hz | Streckeisen STS-2/N seismometer | 2005-12-06 -  | Lat: 50.0, Lng: 12.1'
u'BW.MANZ..EHE | 200.00 Hz | Streckeisen STS-2/N seismometer | 2005-12-06 -  | Lat: 50.0, Lng: 12.1'

Second list contains 2 additional elements.
First extra element 6:
u'BW.MANZ..EHN | 200.00 Hz | Streckeisen STS-2/N seismometer | 2005-12-06 -  | Lat: 50.0, Lng: 12.1'

  [u'Networks:',
   u'BW (BayernNetz)',
   u'Stations:',
   u'BW.MANZ (Manzenberg,Bavaria, BW-Net)',
   u'Channels:',
+  u'BW.MANZ..EHE | 200.00 Hz | Streckeisen STS-2/N seismometer | 2005-12-06 -  | Lat: 50.0, Lng: 12.1',
+  u'BW.MANZ..EHN | 200.00 Hz | Streckeisen STS-2/N seismometer | 2005-12-06 -  | Lat: 50.0, Lng: 12.1',
   u'BW.MANZ..EHZ | 200.00 Hz | Streckeisen STS-2/N seismometer | 2005-12-06 -  | Lat: 50.0, Lng: 12.1']

----------------------------------------------------------------------
Ran 2 tests in 0.064s

FAILED (failures=1)

```
Somehow my method is changing the state of other tests.
I can't track down what I'm doing to the state that is getting across individual tests.
Possibly io.BytesIO  or native / future string namespace getting trounced?

### CODE REVIEW

File: obspy/io/xseed/parser.py [hunk 1]
```diff
@@ -670,6 +671,155 @@ def write_resp(self, folder, zipped=False):
...
+            # This is not correct for more than rdseed -R, although it will parse
+            # Also will not separate stations blockettes by station
+            if record_type == 'S':
+                root_attribute.append(blockette_obj)
+            elif record_type == 'V':
+                self.temp['volume'].append(blockette_obj)
+            elif record_type == 'A':
+                self.temp['abbreviations'].append(blockette_obj)
+
+            # This line breaks other tests: test_string and test_rotation_to_zne
```
Comment (dsentinel) [inline obspy/io/xseed/parser.py]: if `raise Exception` here the following tests will succeed

**Comment by dsentinel** [comment]
This issue was found and working #1185

**Comment by megies** [comment]
> I can't track down what I'm doing to the state that is getting across individual tests.

You can use pytest with "random" plugin to find out which tests interact.

File: obspy/io/xseed/parser.py [hunk 2]
```diff
@@ -670,6 +671,155 @@ def write_resp(self, folder, zipped=False):
...
+                            resp_value += '~'
+                        # print(RESPvalue)
+                        # Lookup if abbv
+                        resp_value = abbv_lookup.get(resp_value, resp_value)
+                        data_resp_value = io.BytesIO(resp_value.encode('utf-8'))
+                        if (hasattr(bfield, 'length') and
+                                bfield.length < len(resp_value)):
+                            # RESP does not use the same length for floats
+                            # as SEED does
+                            bfield.length = len(resp_value)
```
Comment (dsentinel) [inline obspy/io/xseed/parser.py]: bfield is an instance, BUT is defined in a class variable of blockette. Changing this changes the class and all instances of it.

**Comment by dsentinel** [comment]
The joys of python's scoping.
Case closed,  bug in comment above.

**Comment by megies** [comment]
> Case closed, bug in comment above.

I can't follow, was there no bug or why did you close this?

**Comment by dsentinel** [comment]
Sorry for the lack of explanation. I closed the issue to prevent anyone from spending any time on it as I'd found the cause.

No bug in obspy. It was how I was using xseed blockette classes. What I did was rather hackish.
The problem can be avoided if I use deepcopy (copy may even bee sufficient)

<parser.py line:797> I assign a new value to blockette field lengths for RESP fields that are a different length than in corresponding seed or xseed usually floats.
The problem is that these blockette fields are class variables of the blockette classes, not instance variables.  I was changing class variables for all current and future blockette instances.

**Comment by megies** [comment]
Ok, good to hear you found your problem.

**Pull request closed without merging**